scoring/score_submission.py

```diff
@@ -0,0 +1,198 @@
+    },
+    'imagenet_vit_pytorch': {
+        'validation_target': ImagenetVitWorkload().validation_target_value,
+        'test_target': ImagenetVitWorkload().test_target_value,
+        'validation_metric': 'validation/accuracy',
+        'test_metric': 'test/accuracy',
+    },
+    'librispeech_conformer_jax': {
+        'validation_target':
+            LibriSpeechConformerWorkload().validation_target_value,
+        'test_target':
+            LibriSpeechConformerWorkload().test_target_value,
+        'validation_metric':
+            'validation/wer',
+        'test_metric':
+            'test/wer',
+    },
+    'librispeech_conformer_pytorch': {
+        'validation_target':
+            LibriSpeechConformerWorkload().validation_target_value,
+        'test_target':
+            LibriSpeechConformerWorkload().test_target_value,
+        'validation_metric':
+            'validation/wer',
+        'test_metric':
+            'test/wer',
+    },
+    'librispeech_deepspeech_jax': {
+        'validation_target':
+            LibriSpeechDeepSpeechWorkload().validation_target_value,
+        'test_target':
+            LibriSpeechDeepSpeechWorkload().test_target_value,
+        'validation_metric':
+            'validation/wer',
+        'test_metric':
+            'test/wer'
+    },
+    'librispeech_deepspeech_pytorch': {
+        'target':
+            LibriSpeechDeepSpeechWorkload().validation_target_value,
+        'validation_target':
+            LibriSpeechDeepSpeechWorkload().validation_target_value,
+        'test_target':
+            LibriSpeechDeepSpeechWorkload().test_target_value,
+        'validation_metric':
+            'validation/wer',
+    },
+    'ogbg_jax': {
+        'validation_target': OgbgWorkload().validation_target_value,
+        'test_target': OgbgWorkload().test_target_value,
+        'validation_metric': 'validation/mean_average_precision',
+        'test_metric': 'test/mean_average_precision',
+    },
+    'ogbg_pytorch': {
+        'validation_target': OgbgWorkload().validation_target_value,
+        'test_target': OgbgWorkload().test_target_value,
+        'validation_metric': 'validation/mean_average_precision',
+        'test_metric': 'test/mean_average_precision',
+    },
+    'wmt_jax': {
+        'validation_target': WmtWorkload().validation_target_value,
+        'test_target': WmtWorkload().test_target_value,
+        'validation_metric': 'validation/bleu',
+        'test_metric': 'test/bleu',
+    },
+    'wmt_pytorch': {
+        'validation_target': WmtWorkload().validation_target_value,
+        'test_target': WmtWorkload().test_target_value,
+        'validation_metric': 'validation/bleu',
+        'test_metric': 'test/bleu',
+    },
+}
+
+
+def main(_):
+  df = scoring_utils.get_experiment_df(FLAGS.experiment_path)
+  results = {
+      FLAGS.submission_tag: df,
+  }
+  performance_profile_df = scoring.compute_performance_profiles(
+      results,
+      workload_metadata,
```

Comment:
ideally `compute_performance_profiles` would be able to be called from a script (like if a user is using it in a pipeline), and it is a pain to have to supply `workload_metadata` organized like this. ideally we could just look up that metadata given the workload name. this would be possible today but it's annoyingly not because we don't have the metric names in the spec api, just `has_reached_test_target`. I think we should just add

```python
def target_metric_name(self):
  return 'name'
```

to each workload, and then inside `scoring.py:get_index_that_reaches_target(...)` we look up all the info we need from the workload object. I did this for us in [[this PR]](https://github.com/mlcommons/algorithmic-efficiency/pull/443), if you merge that into your branch then I think this will be much cleaner.